### ISSUE TITLE
Publish as Kodi add-on (Krypton/Leia)

### ISSUE THREAD

**Issue by 24mu13** [allejok96/plugin.video.jwb-unofficial#1]
I agree with your decision to resurrect this useful add-on. I also understand the doubts that the previous author wrote before the cancellation, but in conscience after thinking a while, I realised that the benefits having an add-on like this are greater than possible risks.

So could you please publish to official repo as described here?
https://kodi.wiki/view/Submitting_Add-ons

Thank you,
Samuel

**Comment by allejok96** [comment]
No, I'm afraid not.

But first of all, let me say I think the add-on is great and I use it almost daily. It's nothing wrong with it, and I understand why people would want it in the repo. But I don't think it should be there, and I don't see the big problem with that. It's not that hard to install, it just requires a little effort. And I like it to stay that way. Why?

**It's made for people interested in JW broadcasting, not for people who simply stumble across it**

I think people should discover JW.org first and then actively search for this add-on and download it if they want. They should get a taste of the real stuff first, before getting served the B quality. If they discover JW broadcasting through this add-on, and it sucks, perhaps it will scare them away, and I'd rather not see that happening.

**It enforces the need for technical skills and thus a better understanding of potential problems**

I don't know if I'll have the time or ability to fix this if it breaks. For the sake of argument, let's say a bunch of not-technically-skilled users installs this from the repo and it breaks... Who do they call? If a technically skilled person had set this up to begin with, he/she would have been their go-to "support". And for a techy person, installing this is a no-brainier. Plus, since there are almost no updates to the code the need for auto update is low.

**Staying "third party" lowers the risk of giving a bad impression**

If the person installing this had to go about downloading it from some weird site on the internet (read github), they probably had their expectations low and are satisfied if it merely plays a video. Those people have better understanding of potential glitches, or worst case, a hack attempt.

**I respect the guy who made this add-on to begin with**

I know he wanted it gone, but he knew someone would still keep it alive. I have done just that. I can't say I've got his "permission", but let's just say he's aware of it, and does not object to it. And just like that guy I can't stop people from forking this further and putting it in the repo. In fact, by keeping the fork on my github I've increased the risk of that scenario already.

...

So, yeah, I don't think the benefit of easier installation outweighs the things listed above. The add-on is already free and available for everyone. But you'll have to make a slight effort to get it. And with doing so, you yourself take full responsibility for any problems.

I hope you get my point and respect my decision. Keep on using JWB unofficial all you want. I'll still be here for a while to fix any problems you may encounter.

**Comment by 24mu13** [comment]
Thank you Alex for the comprehensive explanation.

**Let's do it short**: _I respect your respect for previous author and I will not fork your repo; but please note that when I found it, I was ready to publish the code from original author as you did and - differently from you - also to publish to the Kodi repo_.

For the sake of the topic, let me add something, even more generic, to support my view.

**The risks** (see [README.md](https://github.com/allejok96/plugin.video.jwb-unofficial/blob/de26041e9466f575497a2c98a97c6e785b3fc8d2/README.md))

First of all I read several times the [article](https://wol.jw.org/en/wol/d/r1/lp-e/2018364) that struggled the original author, as well as related articles in our amazing [library](https://wol.jw.org) on this topic.
Basing on that info and especially on the Biblical principles behind, I totally agree with the **first two** points.
Instead, I disagree on the last two:

 - _You are trusting a third-party to deliver spiritual food to you_: it's not true since the **source-code is publicly available**, so anyone could verify the URLs are genuine (as you can do with the links I wrote above, for instance)
 - _This repo could get hacked, you know_: yes, it's theoretically true for anything in _this_ world, at least for now, but again the **open-source** nature of this - let me say, very simple - add-on gives us an easy way to "certify" that is not, at any point in time!

**Reasons why you don't publish** (see previous [comment](https://github.com/allejok96/plugin.video.jwb-unofficial/issues/1#issuecomment-457864705))

As I said at the very beginning I agree with **last point**, and that's enough for me. But let me point for a moment to the **first three** points, that share a common idea...
 - _It's made for people interested [...]_
 - _It enforces the need for technical skills [...]_
 - _Staying "third party" lowers the risk of giving a bad impression_

This add-on is a "third-party add-on" regardless his place on the "official" Kodi repo! If you want to make clear that is absolutely not related with **Watchtower** society, would be better to introduce a dialog/disclaimer that points to the official web-[TV](https://tv.jw.org) every time you start it, instead of avoiding the Kodi repo. 

More in general, this add-on - like the majority of Kodi add-ons - is very simple: you can see them like a **smarter browser**, nothing more! Thus, this add-on is very useful for A) people who like to consume web media-content in a media-centre and consistent way and/or B) people (like my parents) who do not have any of supported HW for official TV app.
So, there is no point - at least for me - to place again this add-on on GitHub without having it on Kodi repo, to quickly install it. If you rather prefer to keep it geeky/technical would be enough **having it as a zip**  and exchanging it via whatever mean (mail/message/etc). When previous author closed it, in fact, I immediately saved the zip for future use.

In conclusion, as I realised many times in the past, often our [organisation](https://www.jw.org) surprised us with a new tech (or app) that covers eventually TV or HW unsupported right now.. it's not so important.... but with this comment I wanted to clarify my view on this topic, basing on what I got from the mentioned articles.

S.

**Comment by allejok96** [comment]
Thanks for sharing your view. Let me just address some things you say and you'll see I understand you perfectly.

*The source-code is publicly available, so anyone could verify the URLs are genuine.*
Yes, nerds as you an me can. Normal people most probably can't. And most people won't. I'm a linux and foss lover. No hard feelings. I don't think the "untrusted thirdparty nature" of this is a problem in reality. I mean, the code is very simple. It just uses the public API. And in the same way you'll trust your browser when visiting JW.org. I just make it *super clear* so there is not a slightest chance anyone can get confused and blame me or the organization in some way.

*Getting hacked is theoretically true for anything in this world*.
I agree. But I don't claim to have the same security standard and 24/7 watch on my accounts as the watchtower society has over it's stuff. A single person with bad motives can still do a lot of damage, even if everything is open source. Just look at the NPM leftpad fiasco. That was no hack, but it still broke a lot of stuff. Again, not a real problem, just making it super clear.

*It would be better to introduce a dialog/disclaimer.*
That's the one change I've already done to the code besides changing the logo.

*You can see them like a smarter browser*
I don't know if you are referring to the fact that most sites forbid stuff like scrappers and that this add-on should be excused since it's basically a browser... But that doesn't worry me. The terms allows for this and I've gotten written permission from the Swedish branch office to use add-ons like this to watch broadcasting.

*It would be enough having it as a zip*.
Github has a download as zip button, so essentially that's already how it works. But GH also has issues, forks, pull requests, comments and all other open source goodness.

Now here comes the main thing where I think our views differ:

*People (like my parents) who do not have any of supported HW for official TV app*.
And me too. And my mum. And my uncle. That doesn't mean I can't just download a zip for them. It will be me setting up their Kodi box anyways. Btw I actually have set up a experimental third party repo for the add-on here on GH to ease potential updates for my friends. It's not in use yet, but you see I don't have anything against repos. The thing is I take responsibility for my family's stuff. I can call them and talk to them if something happens, or they run into trouble. I'm sure you take responsibility over your and your family's stuff in the same way.

I just don't like taking responsibility over how the global userbase of Kodi will experience Jehovah's witnesses broadcasting. I don't see the need for having the add-on easily discoverable in the repo, as I've said I think people who never seen broadcasting should see the real website and JW.org first. And I think most witnesses who can set up a Kodi box are capable of downloading this add-on by them selves. And thus take responsibility for it. They can help their friends and family with any potential problems. They can explain if something weird happens. This way the responsibility is split up on each and every person who chooses to use the add-on.

I don't have any doubts if using add-on is alright or not. I think it's fine. But as you've seen some people don't and so this is a messy greyzone. It's a conscience thing. I'm doing this to stay out of that zone and avoid future problems. I force each and everyone to make their own decision whether they think this is ok or not. I don't promote the add-on, I'm very clear on what potential downsides there is. The positive sides need not be mentioned, as they are obvious.

You may think I'm a bit extreme by my choice of installation method, but it's just my way of keeping my good conscience. If your conscience tells you different, that's fine. And by all means, if you want to publish the add-on, don't hesitate to fork my repo. I rather see the small improvements I've made go into the new add-on then just having someone upload the old one again. (Just beware that there is a bug with the disclaimer message I added).

**Comment by ca0abinary** [comment]
Interesting conversation. I can see both sides, and you've both read my reasoning on the matter. As a historical side-note I had to explain my position a couple times to the Kodi official repo folks who didn't understand my reasoning clearly at first either.

Each person has a different conscience and I respect the balance @allejok96 has made here, particularly with getting approval from the local branch. As brought out in [w97 8/1 p. 4 "How to Train Your Conscience"](https://wol.jw.org/en/wol/d/r1/lp-e/1997561#h=2:0-2:149) “A CLEAN conscience is the best pillow.” ([1 Peter 3:16](https://wol.jw.org/en/wol/b/r1/lp-e/nwtsty/E/2019/60/3#study=discover&v=60:3:16))
